### DISCUSSION TITLE
Aggregates in Detail List

### DISCUSSION THREAD

**Discussion by rhemasystems** [serenity-is/Serenity#6656] [Help] [answered]
I have a detail grid in a master dialog, how can i add row numbering for the grid and also total rows in th grid footer

![image](https://user-images.githubusercontent.com/50826372/212287286-89a153e6-a6b0-4c06-9889-0ebb3d743899.png)

**Comment by rhemasystems** [comment]
Please any help will be greatly appreciated

**Comment by wezmag** [comment, marked as answer]
You can use SQL `ROW_NUMBER` to add row numbers.
```
[Expression("ROW_NUMBER() OVER(ORDER BY MasterTableId)")]
public int? RowNumber
{
    get => fields.RowNumber[this];
    set => fields.RowNumber[this] = value;
}
```
You might need to disable sorting on your grid.
For total rows, just use `Aggregators.Max`.
https://github.com/serenity-is/common-features/blob/master/src/Serenity.Demo.BasicSamples/Modules/Grids/GroupingAndSummariesInGrid/GroupingAndSummariesInGrid.ts

**Reply by rhemasystems** [reply]
Please how can I achieve at client side, as I add new rows to the grid then it auto number them.
Thank you
